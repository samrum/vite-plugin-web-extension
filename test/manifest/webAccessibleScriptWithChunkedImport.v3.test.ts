import { getResourceDir, runManifestV3Test } from "./manifestTestUtils";

const resourceDir = getResourceDir("webAccessibleScriptWithChunkedImport");

runManifestV3Test("webAccessibleScriptWithChunkedImport", () => ({
  web_accessible_resources: [
    {
      resources: [`${resourceDir}/webAccessibleScript1.js`],
      matches: ["<all_urls>"],
    },
    {
      resources: [`${resourceDir}/webAccessibleScript2.js`],
      matches: ["https://*/*", "http://*/*"],
    },
  ],
}));
