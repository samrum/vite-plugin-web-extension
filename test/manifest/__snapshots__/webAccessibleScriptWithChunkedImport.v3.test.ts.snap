// Vitest Snapshot v1

exports[`webAccessibleScriptWithChunkedImport - Manifest V3 1`] = `
[
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"webAccessibleScript1\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1.js",
    "fileName": "assets/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1.js": {
        "code": "log(\\"webAccessibleScript1\\");",
        "originalLength": 63,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 28,
      },
    },
    "name": "test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "import { l as log } from \\"../../../../log.js\\";
log(\\"webAccessibleScript2\\");
",
    "dynamicImports": [],
    "exports": [],
    "facadeModuleId": "vite-plugin-web-extension/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2.js",
    "fileName": "assets/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {
      "assets/log.js": [
        "l",
      ],
    },
    "imports": [
      "assets/log.js",
    ],
    "isDynamicEntry": false,
    "isEntry": true,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2.js": {
        "code": "log(\\"webAccessibleScript2\\");",
        "originalLength": 63,
        "removedExports": [],
        "renderedExports": [],
        "renderedLength": 28,
      },
    },
    "name": "test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "code": "function log(message) {
  console.log(message);
}
export {
  log as l
};
",
    "dynamicImports": [],
    "exports": [
      "l",
    ],
    "facadeModuleId": null,
    "fileName": "assets/log.js",
    "implicitlyLoadedBefore": [],
    "importedBindings": {},
    "imports": [],
    "isDynamicEntry": false,
    "isEntry": false,
    "isImplicitEntry": false,
    "map": null,
    "modules": {
      "vite-plugin-web-extension/test/manifest/resources/shared/log.js": {
        "code": "function log(message) {
  console.log(message);
}",
        "originalLength": 65,
        "removedExports": [],
        "renderedExports": [
          "default",
        ],
        "renderedLength": 49,
      },
    },
    "name": "log",
    "referencedFiles": [],
    "type": "chunk",
    "viteMetadata": {
      "importedAssets": Set {},
      "importedCss": Set {},
    },
  },
  {
    "fileName": "test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2.js",
    "name": undefined,
    "source": "(async()=>{await import(chrome.runtime.getURL(\\"assets/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2.js\\"))})();",
    "type": "asset",
  },
  {
    "fileName": "manifest.json",
    "name": undefined,
    "source": "{
  \\"version\\": \\"1.0.0\\",
  \\"name\\": \\"Manifest Name\\",
  \\"manifest_version\\": 3,
  \\"web_accessible_resources\\": [
    {
      \\"resources\\": [
        \\"test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1.js\\",
        \\"assets/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript1.js\\",
        \\"assets/log.js\\"
      ],
      \\"matches\\": [
        \\"<all_urls>\\"
      ]
    },
    {
      \\"resources\\": [
        \\"test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2.js\\",
        \\"assets/test/manifest/resources/webAccessibleScriptWithChunkedImport/webAccessibleScript2.js\\",
        \\"assets/log.js\\"
      ],
      \\"matches\\": [
        \\"https://*/*\\",
        \\"http://*/*\\"
      ]
    }
  ]
}",
    "type": "asset",
  },
]
`;
